File: SWIG-C/example_main_as_class.py
#!/usr/bin/pypy

import c_example   # native C functions


if __name__ == '__main__':
    # wrap into class like behavior
    class Example(object):
        def __init__(self, *arg):
            self.descr = c_example.Example(arg[0])

        def Get(self, *arg):
            return c_example.Get(self.descr)

        #def __del__(self):
        #   close(), free() and other
    # end of wrap

    a = Example('teszt-A')
    b = Example('teszt-B')

    print a.Get()
    print b.Get()
    print a.Get()
    print a.Get()
    print a.Get()
    print a.Get()
    print b.Get()

File: SWIG-C/example_main_as_class.cs
using System;

namespace CExample
{

    public class Example
    {
        private int descr;
        public Example(string s)
        {
            this.descr = c_example.Example(s);
        }

        public string Get()
        {
            return c_example.Get(this.descr);
        }
    }

    public class Example_main
    {
        public static void Main(string[] args)
        {
            Example a = new Example("Teszt-A");
            Example b = new Example("Teszt-B");

            Console.WriteLine(a.Get());
            Console.WriteLine(b.Get());
            Console.WriteLine(a.Get());
            Console.WriteLine(a.Get());
            Console.WriteLine(a.Get());
            Console.WriteLine(a.Get());
            Console.WriteLine(b.Get());
        }
    }
}
